### PR TITLE
XSUP-41319/CS Falcon - fixed duplications in IDP Detections

### PR DESCRIPTION
**Pull request by RotemAmit**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/XSUP-41319)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by MLainer1 — COMMENTED** [review]
🆒

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 1]
```diff
@@ -741,6 +741,27 @@ def incident_to_incident_context(incident):
...
+        Args:
+            detection (dict): the detection.
+            time_key (str): the key of the wanted date&time field.
+    """
+    demisto.debug(f'fix_time_field {time_key=}')
+    str_date = detection[time_key]
+    split_date = str_date.split('.')
+    relevant_microseconds = split_date[1][:6]
+    # if 'Z' isn't in relevant_microseconds it means that it was removed since there was more than 5 digits in the microseconds.
+    fixed_date = f'{split_date[0]}.{relevant_microseconds}Z' if 'Z' not in relevant_microseconds else str_date
```
Comment (eyalpalo) [inline Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:760]: If we get a timestamp with 6 digits (which according to the docstring is the minimal amount of digits) we will remove the Z as well. In which situation, we will have the Z after taking the first 6 chars?

**Review by eyalpalo — CHANGES_REQUESTED** [review]

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py (hunk 1 above)
Comment (RotemAmit) [reply Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:760]: We will have the Z after taking the first 6 chars if there are 5 digits or less in in the microseconds.

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 2]
```diff
@@ -752,6 +773,10 @@ def detection_to_incident_context(detection, detection_type, start_time_key: str
         :rtype ``dict``
     """
     add_mirroring_fields(detection)
+    demisto.debug(f'detection_to_incident_context, {detection_type=}')
+    if detection_type == IDP_DETECTION_FETCH_TYPE:
+        demisto.debug(f'detection_to_incident_context, {detection_type=} calling fix_time_field')
+        fix_time_field(detection, start_time_key)
```
[the remark this answers is not in the document]
Comment (RotemAmit) [reply Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:779]: Yes

**Pull request merged**